src/feature.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  EvaluationApi,
+  EvaluationBatchRequest,
+  EvaluationBatchRequestFlagTagsOperatorEnum,
+  EvaluationEntity,
+  createConfiguration,
+  ServerConfiguration,
+} from "flagr-client";
+
+export interface FlagCallbacks<T> {
+  [key: string]: (attachment?: JsonObject) => T | Promise<T>;
+}
+
+declare type Json = string | number | boolean | null | JsonArray | JsonObject;
+
+declare type JsonArray = Array<Json>;
+
+declare type JsonObject = {
+  [key: string]: Json;
+};
+
+export interface Variant {
+  key: string | null;
+  attachment: JsonObject | null;
+}
+
+const nullVariant: Variant = {
+  key: null,
+  attachment: null,
+};
+
+export interface Config {
+  flagrUrl: string;
+  tags?: string[];
+  tagOperator?: EvaluationBatchRequestFlagTagsOperatorEnum;
+}
+
+export function createFeature(config: Config) {
+  return new Feature(config);
+}
+
+export class Feature {
+  private context: JsonObject = {};
+  private evaluationResults: Map<string, Variant> = new Map();
+  private api: EvaluationApi;
+
+  constructor(private _config: Config) {
+    this.api = new EvaluationApi(
+      createConfiguration({
+        baseServer: new ServerConfiguration<{}>(
+          `${this._config.flagrUrl}/api/v1`,
+          {}
+        ),
+      })
+    );
+  }
+
+  setContext(context: JsonObject) {
+    this.context = context;
+    this.evaluationResults = new Map();
+  }
+
+  addContext(context: JsonObject) {
+    this.setContext({ ...this.context, ...context });
+  }
+
+  async match(flag: string, matchVariant: string = "on") {
+    const callbacks: FlagCallbacks<boolean> = { otherwise: async () => false };
+    callbacks[matchVariant] = async () => true;
+    return this.evaluate(flag, callbacks);
+  }
+
+  async evaluate(flag: string, callbacks: FlagCallbacks<any>) {
```

Comment:
Gotta pass through the generic

```suggestion
  async evaluate<T>(flag: string, callbacks: FlagCallbacks<T>) {
```

Reply:
Ah. It all makes sense now